java/java.lsp.server/vscode/src/extension.ts

```diff
@@ -65,33 +65,112 @@ function findClusters(myPath : string): string[] {
     return clusters;
 }
 
-export function activate(context: ExtensionContext) {
-    //verify acceptable JDK is available/set:
-    let specifiedJDK = workspace.getConfiguration('netbeans').get('jdkhome');
-    const beVerbose : boolean = workspace.getConfiguration('netbeans').get('verbose', false);
-    let info = {
-        clusters : findClusters(context.extensionPath),
-        extensionPath: context.extensionPath,
-        storagePath : context.globalStoragePath,
-        jdkHome : specifiedJDK,
-        verbose: beVerbose
-    };
-    
-    let log = vscode.window.createOutputChannel("Java Language Server");
+function findJDK(onChange: (path : string | null) => void): void {
+    function find(): string | null {
+        let nbJdk = workspace.getConfiguration('netbeans').get('jdkhome');
+        if (nbJdk) {
+            return nbJdk as string;
+        }
+        let javahome = workspace.getConfiguration('java').get('home');
+        if (javahome) {
+            return javahome as string;
+        }
 
-    vscode.extensions.all.forEach((e, index) => {
+        let jdkHome: any = process.env.JDK_HOME;
+        if (jdkHome) {
+            return jdkHome as string;
+        }
+        let jHome: any = process.env.JAVA_HOME;
+        if (jHome) {
+            return jHome as string;
+        }
+        return null;
+    }
+
+    let currentJdk = find();
+    workspace.onDidChangeConfiguration(params => {
+        if (!params.affectsConfiguration('java') && !params.affectsConfiguration('netbeans')) {
```

Comment:
I would rather check on full property name: `java.home` and `netbeans.jdkhome`...

Reply:
`ConfigurationChangeEvent` doesn't seem to have a way to check that.